Dockerfile_build

```diff
@@ -1,48 +1,53 @@
 
+# "Wait," you ask. "What's going on here?" Rather than handling rustup validation
+# and verification, we can list the rust container as a prior build stage, and
+# then pull in the artifacts we need. There is an added benefit that tagged versions
+# also include minory releases, so 1.2 includes 1.2.1 and so on, for bugfix releases.
+FROM rust:1.52 as RUSTBUILD
+
 FROM golang:1.16
 
 # Install common packages
 RUN apt-get update && \
     apt-get install --no-install-recommends -y \
-    gcc-arm-linux-gnueabihf libc6-dev-armhf-cross \
-    gcc-aarch64-linux-gnu libc6-dev-arm64-cross \
-    libtinfo5 \
-    ruby \
+    openssl libtinfo5 ruby \
     ca-certificates curl file gnupg \
     build-essential cmake \
     libxml2-dev libssl-dev zlib1g-dev \
     autoconf automake autotools-dev libtool xutils-dev valgrind && \
     rm -rf /var/lib/apt/lists/*
 
-# Download, verify, and install ragel version 6. 
-ENV RAGEL_VERSION=6.10
-RUN curl https://www.colm.net/files/thurston.asc | gpg --import - && \
-    curl https://www.colm.net/files/ragel/ragel-${RAGEL_VERSION}.tar.gz -O && \
-    curl https://www.colm.net/files/ragel/ragel-${RAGEL_VERSION}.tar.gz.asc -O && \
-    gpg --verify ragel-${RAGEL_VERSION}.tar.gz.asc ragel-${RAGEL_VERSION}.tar.gz && \
-    tar -xzf ragel-${RAGEL_VERSION}.tar.gz && \
-    cd ragel-${RAGEL_VERSION}/ && \
-    ./configure --prefix=/usr/local && \
-    make && \
-    make install && \
-    cd .. && rm -rf ragel-${RAGEL_VERSION}*
+# Install rust and rust tooling
+COPY --from=RUSTBUILD /usr/local/cargo /usr/local/cargo
+COPY --from=RUSTBUILD /usr/local/rustup /usr/local/rustup
+
+ENV RUSTUP_HOME=/usr/local/rustup \
+    CARGO_HOME=/usr/local/cargo \
+    PATH=/usr/local/cargo/bin:/usr/local/bin:/usr/local/ragel7/bin:$PATH
+RUN rustup component add rustfmt clippy && \
+    # Use sccache rustc wrapper for friendly build caching
+    cargo install sccache && \
+    cargo install wasm-pack && \
+    rustup component add rust-std --target wasm32-unknown-unknown
+# This is defined separately, so it doesn't attempt to use sccache until it is
+# actually installed.
+ENV RUSTC_WRAPPER=sccache
 
-# Download, verify, and install ragel version 7. This version depends on colm,
-# so we get that as well. Already have the gpg key above.
+# Install additional tooling and requirements for building flux.
 ENV COLM_VERSION=0.14.2
-RUN curl https://www.colm.net/files/colm/colm-${COLM_VERSION}.tar.gz -O && \
+ENV RAGEL7_VERSION=7.0.1
+RUN curl https://www.colm.net/files/thurston.asc | gpg --import - && \
+    curl https://www.colm.net/files/colm/colm-${COLM_VERSION}.tar.gz -O && \
     curl https://www.colm.net/files/colm/colm-${COLM_VERSION}.tar.gz.asc -O && \
     gpg --verify colm-${COLM_VERSION}.tar.gz.asc colm-${COLM_VERSION}.tar.gz && \
     tar -xzf colm-${COLM_VERSION}.tar.gz && \
     cd colm-${COLM_VERSION}/ && \
     ./configure --prefix=/usr/local/ragel7 --disable-manual && \
```

Comment:
I notice you're still using the `ragel7` prefix. That works for me, but I just want to mention that ragel 7 is intended to be backwards compatible with 6, so it is *okay* to replace 6 with 7 by installing to the same location., which was `/usr/local/`. The unique prefix was just so they could co-exist. But I see cargo is also a unique prefix ...

Reply:
I was hoping you'd say something about that. I did try to take the prefix out, but it was causing build issues that I just didn't want to chase much more at the expense of the good in this patch.

Reply:
Do you remember the nature of the issues? Like building ragel, or building flux?

Reply:
Building ragel, with some interaction with colm, I believe. I didn't look too deeply at it, and it's probably just a pathing issue. It was more that building the image takes ~20 minutes, so poking at it could still take me another hour at its simplest to fix, and I just didn't want to devote it.

Reply:
Oh possibly `colm` wasn't available to build `ragel`, or something like that. Things are all good as they are. Not the worst to segregate the applications.